mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -547,25 +538,6 @@ class Completions(
       else (buf.result, SymbolSearch.Result.COMPLETE)
 
     end filterInteresting
-
-    private def isNotPackageObject(sym: Symbol) =
-      !sym.isPackageObject
-
-    private def isNotAModuleOrModuleIsValidForPos(sym: Symbol) =
-      !sym.info.typeSymbol.is(
-        Flags.Module
-      ) || cursorPositionCondition.isObjectValidForPos
-
-    private def isNotAMethodOrMethodIsValidForPos(sym: Symbol) =
-      cursorPositionCondition.isMethodValidForPos || !sym.is(
-        Flags.Method
-      ) // !sym.info.typeSymbol.is(Flags.Method) does not detect Java methods
-
-    private def isNotClassOrTraitOrTheyAreValidForPos(sym: Symbol) =
-      cursorPositionCondition.isClassOrTraitValidForPos ||
-        sym.is(Flags.Method) ||
-        !sym.isClass && !sym.info.typeSymbol.is(Trait)
```

Comment:
This was not correct because symbol might be at least also a ValDef.
Also, my suggestion of checking if it's a trait over a `.info` was wrong :facepalm: 
```scala
def foo(param: java.util.List[A]): Unit =
  para@@ // <- here param is not a Method
        // <- also `sym.info.typeSymbol.is(Trait) == true` here because the of param is `java.util.List` 
```

Reply:
This method is only excluding class or trait where they are not valid for pos. It does not affect `ValDef`

Reply:
Looks like it's explained by @dos65  : `sym.info.typeSymbol.is(Trait) == true`